sidecar/src/watchdog.rs

```diff
@@ -46,12 +46,44 @@ impl Watchdog {
         let mem_usage_bytes = Arc::new(AtomicUsize::new(0));
         let handle_mem_usage_bytes = mem_usage_bytes.clone();
 
+        let still_alive = Arc::new(AtomicU32::new(0));
+        let still_alive_thread = still_alive.clone();
+
+        let interval = self.interval.period();
+        std::thread::spawn(move || {
+            let mut repeats = false;
+            let mut last = 0;
+            loop {
+                std::thread::sleep(interval);
+                let current = still_alive_thread.load(Ordering::Relaxed);
+                if last != current {
+                    if current == u32::MAX {
```

Comment:
A comment about this constant or naming it `SHUTDOWN` would make things clearer.